.gitignore

```diff
@@ -20,3 +20,9 @@ models/px4flow/px4flow.sdf
 models/r200/r200.sdf
 models/sf10a/sf10a.sdf
 models/standard_vtol/standard_vtol.sdf
+models/temp*
+worlds/temp*
```

Comment:
Why do we need a temp directory? Can we just generate temporary files into `/tmp`?

Reply:
For SITL we still generate models into /tmp, for HITL we generate into models/temp_*, the worlds for both HITL and SITL are generated on a configuration requirement basis.

Reply:
Can we generate files into the same place for both `sitl` and `hitl`?

Reply:
I agree with @Jaeyoung-Lim suggestion here

Reply:
So this isn't actually a temp directory, it's the prefix for user generated worlds and ONLY HITL generated models.

Reply:
@bperseghetti What is the benefit of having this inside the repository, and not somewhere like `/tmp` ?

Reply:
It doesn't work in /tmp for HITL when having to launch it with "gazebo ____.world method" instead of the client and server separately with other "gz model --spawn-file_______.sdf --model-name _____ method" and there is no reason to put these in /tmp. 

After all .gitignore exists for this very purpose. I mean, same question could be asked about why we spawn iris.sdf into models/iris instead of only ever /tmp. Unless you think that method should be removed as well and there shouldn't be an iris.sdf in models/iris?

Reply:
@bperseghetti So by adding it to the gitignore, and using it as the model directory we are trying to hide that the model is generated and since we don't want to copy in all model.config file we added it to gitignore as a compromise. I would really like to remove this approach if possible

If generated models are temporary, what benefit does it give to have them in a `temp` directory inside the repository? This will make the models persistent in the repo and that is why I think putting it in somewhere like `/tmp` is better. This also provides a clean separation between static models and dynamically generated models.

Reply:
Ok, this was discussed on dev call and I think we all agree it is not a blocker.

Reply:
I think there was a misunderstanding - I meant in the dev call that having world file generated to support HITL was not a blocker. I am still not sure what benefit we have by generating files into the repo that are going to be ignored